components/director/internal/domain/formation/service.go

```diff
@@ -362,6 +364,12 @@ func (s *service) assign(ctx context.Context, tnt, objectID string, objectType g
 	if err != nil {
 		return nil, errors.Wrapf(err, "while getting formation %q", formation.Name)
 	}
+	// TODO Remove default scenario check after removing default scenario
+	if formation.Name != model.DefaultScenario && objectType == graphql.FormationObjectTypeApplication {
```

Comment:
Change these 3 if's(lines 368, 373, 378) into a switch statement where the cases will be the different object types. And after we remove the `DefaultScenario` the code `switch` will look even more readable. Also, this `switch` can be extracted into a function.